pkg/discovery/fsm/fsm_test.go

```diff
@@ -127,6 +127,44 @@ var _ = Describe("FSM", func() {
 		})
 	})
 
+	Context("when individual state timeout is set", func() {
+		It("override default timeout and transition to another state when the timeout is reached", func() {
+			respCh := make(chan string)
+			respond := func(interface{}) error {
+				respCh <- "timeout"
+				return nil
+			}
+			trs := []*Transition{
+				WhenIn("Init").GotEvent("StartTest").GoTo("AwaitTimeout").WithTimeout(5 * time.Millisecond),
+			}
+			cbs := []*Callback{
+				WhenStateTimeout().Do(respond),
+			}
+			callbacks := map[string][]*Callback{}
+			transitions := map[TransitionID]*Transition{}
+			for _, c := range cbs {
```

Comment:
Is there a reason for using a loop here (and below)? These are single element slices that you iterator over.

Reply:
due to copy/paste - updated the code for this newly added test